src/cargo/core/workspace.rs

```diff
@@ -1712,63 +1712,132 @@ impl InheritableFields {
         }
     }
 
-    pub fn dependencies(&self) -> Option<BTreeMap<String, TomlDependency>> {
-        self.dependencies.clone()
+    pub fn dependencies(&self) -> CargoResult<BTreeMap<String, TomlDependency>> {
+        self.dependencies.clone().map_or(
+            Err(anyhow!("[workspace.dependencies] was not defined")),
+            |d| Ok(d),
+        )
     }
 
-    pub fn version(&self) -> Option<semver::Version> {
-        self.version.clone()
+    pub fn get_dependency(&self, name: &str) -> CargoResult<TomlDependency> {
+        self.dependencies.clone().map_or(
+            Err(anyhow!("[workspace.dependencies] was not defined")),
+            |deps| {
+                deps.get(name).map_or(
+                    Err(anyhow!(
+                        "dependency {} was not found in [workspace.dependencies]",
+                        name
+                    )),
+                    |dep| Ok(dep.clone()),
+                )
+            },
+        )
     }
 
-    pub fn authors(&self) -> Option<Vec<String>> {
-        self.authors.clone()
+    pub fn version(&self) -> CargoResult<semver::Version> {
+        self.version
+            .clone()
+            .map_or(Err(anyhow!("[workspace.version] was not defined")), |d| {
```

Comment:
Using `[]` makes sense for `dependencies` but not as much for regular values like these.  Maybe we should consistently just do back-ticks for all of them including `dependencies`?

Reply:
I will do whatever works best, I think I only used `[]` due to past attempts using it/seeing it before. If you think using backticks and making it \`workspace.version\` is better across the board we will do that.